New ad network: sunmedia

operezes opened the pull request:
Hi Team 
We are interested to integrate our ad plugin (sunmedia) in AMP

——
Example:
http://static.addevweb.com/SMVpaidCreatives/amp/examples/ads.amp.max.html

* You have to scroll until you watch the ad in the middle of text block
——

I have signed CLA doc as operezes@yahoo.es, and in GitHub  I am using same email account.

If you need any additional information or whatever from my side feel free to ask me 

Best regards
Oscar

Closes https://github.com/ampproject/amphtml/issues/7426.

lannka commented:
@operezes would you follow the [guide](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request) and implement render-start and no-content APIs?

operezes commented:
Hi Iannka, nice to meet you

I had implemented render-start inside my js plugin, which is loaded after AMP loads main sunmedia.js. no-content was not implemented, but I already have added it to that js:

https://vod.addevweb.com/sunmedia/amp/ads/SMIntextAMP.js

render-sart - line 465
no-content: line 199

Thank you very much
 Oscar

lannka commented:
@operezes in that case, you will need add `renderStartImplemented: true,` to the config. otherwise the events are ignored.

operezes commented:
Hi Iannka, I already have added that parameter (renderStartImplemented: true) in config.js file

Best regards
Oscar

lannka commented:
Please add your example to `ads.amp.html`. Also pls update `amp-ad.md`

operezes commented:
Hi Iannka, I already have updated ads.amp.html and amp-ad.md

Thank you very much
Oscar

lannka commented:
@operezes did you make your example work locally?
I got a console error:
> Uncaught TypeError: Cannot read property 'cid' of undefined
>    at SMIntext.init (SMIntextAMP.js:16)
>   at window.smintextstart (SMIntextAMP.js:510)
>   at HTMLVideoElement.<anonymous> (SMIntextAMP.js:498)

Check [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request) if you don't know how to run it locally.

operezes commented:
Hi Iannka

Sorry for the mess here, we had created other fully local testing environment.
We have updated integration following all instructions described in "amphtml" doc.

We have updated:

/ads/sunmedia.js
/ads/sunmedia.md
/ads/_config.js
/3p/integration.js
/extensions/amp-ad/amp-ad.md
/examples/ads.amp.html

You can check testing integration here, running gulp server:
http://localhost:8000/examples/ads.amp.max.html?type=sunmedia

We have forced one ad for that host

Thank you very much and sorry for the confusion
Oscar

lannka commented on 3p/frame.max.html:
```diff
@@ -6,7 +6,17 @@
 <body style="margin:0">
 <div id="c" style="position:absolute;top:0;left:0;bottom:0;right:0;">
-  <script>draw3p()</script>
+  <script>
```
can you revert changes here?

lannka replied:
please revert the file

lannka commented on 3p/remote.html:
```diff
@@ -4,23 +4,25 @@
-if (!(/^\d+(-canary)?$/.test(v))) return;
-var u = 'https://3p.ampproject.net/'+encodeURIComponent(v)+'/f.js';
-document.write('<script'+' src="'+encodeURI(u)+'"><'+'/script>');
+//var v = location.search.substr(1);
```
here as well

lannka commented on examples/ads.amp.html:
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
+  <amp-ad width="300" height="1"
+      type="sunmedia"
+      layout="responsive"
+      data-cskp=1
```
double quote the value

lannka commented on examples/ads.amp.html:
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
+      layout="responsive"
+      data-cskp=1
+      data-cid="sunmedia_test"
+      data-crst=1>
```
same here

lannka commented on 3p/remote.html:
```diff
@@ -14,13 +14,14 @@
 <div id="c" style="position:absolute;top:0;left:0;bottom:0;right:0;">
   <script>
-    draw3p(undefined,
+    /*draw3p(undefined,
```
revert the file

operezes commented:
Hi Iannka, sorry for the confusion here, I already have left the files as  original project

lannka commented on 3p/frame.max.html:
```diff
@@ -10,4 +10,4 @@
 <script>if (window.docEndCallback) window.docEndCallback()</script>
 </body>
-</html>
+</html>
```
please revert

lannka commented on 3p/remote.html:
```diff
@@ -25,4 +25,4 @@
 <script>if (window.docEndCallback) window.docEndCallback()</script>
 </body>
-</html>
+</html>
```
please revert

lannka commented on ads/sunmedia.js:
```diff
@@ -0,0 +1,35 @@
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+/**
```
nit: add a blank line below import

lannka commented on ads/sunmedia.js:
```diff
@@ -0,0 +1,35 @@
+    global._sunmedia_amp.mandatory_data, global._sunmedia_amp.allowed_data);
+
+  loadScript(global, 'https://vod.addevweb.com/sunmedia/amp/ads/SMIntextAMP.js');
+
```
nit: remove blank line here

lannka commented on examples/ads.amp.html:
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
+      data-cskp="1"
+      data-cid="sunmedia_test"
+      data-crst="1">
+    <div placeholder></div>
```
please sync to the latest code and remove the placeholder and fallback divs

lannka commented:
please also fix the Travis build

lannka commented on examples/ads.amp.html:
```diff
@@ -1158,6 +1159,15 @@ <h2 id="sovrn">SOVRN</h2>
   </amp-ad>
 
+  <h2 id="sunmedia">SunMedia</h2>
+  <amp-ad width="300" height="1"
```
please change height to a bigger value to avoid a resize. for e.g.: `250`

operezes commented:
Hi Iannka,

I have watched you have merged our plugin  and  I can see the plugin included in the project, but I am trying to check the plugin in a real test page, and I get the next error message: 

Type "sunmedia" is not supported in amp-ad

Should I do anything more to be approved by AMP to be able using sunmedia.js in live environment?

Thank you for your help
Oscar

lannka commented:
@operezes you will need to wait a release cycle to be able to test it on live environment, which should be next Thursday.

operezes commented:
Hi Iannka, ok, thank you very much for your help :-)
Oscar
